src/Player.js

```diff
@@ -39,23 +39,22 @@ class Player extends Character {
 
   /**
    * Convert prompt tokens into actual data
-   * @param {string} prompt
-   * @param {object} data
+   * @param {function} promptBuilder (data) => prompt template
+   * @param {object} extraData (key is used by promptBuilder)
    */
-  interpolatePrompt(prompt, data) {
-    data = Object.assign(
-      data || {},
-      this.getAttributes()
-    );
-
-    for (const key in data) {
-      prompt = prompt.replace('%' + key, data[key]);
-    }
-
-    // replace any unknown tokens
-    prompt = prompt.replace(/%([a-z_]+?)/, '[BAD: $1]');
-
-    return prompt;
+  interpolatePrompt(promptBuilder, extraData = {}) {
+    const buildAttributeStr = attr => `${this.getAttribute(attr)}/${this.getMaxAttribute(attr)}`;
```

Comment:
Yeah, I put a comment regarding that... I do like the idea of having the prompt built by a function, but agree that this is less flexible for the time being. I'd rather see something where you pass in a list of attributes to be reformatted as `current/max` and the promptBuilder recognizes that.